packages/flutter/lib/src/widgets/_window.dart

```diff
@@ -0,0 +1,855 @@
+  /// The [preferredConstraints] are the constraints placed upon the size
+  /// of the window. This might not be honored by the platform.
+  /// This field enforces a minimum and maximum size on the window. If the
+  /// user attempts to resize the window beyond these constraints, the platform
+  /// will enforce the constraints according to its own policy. For example, the
+  /// platform might clip the content to fit within the resized window, or it might
+  /// prevent the window from being resized altogether. If null, the window wil
+  /// be unconstrained.
+  ///
+  /// If the [preferredSize] is null, then the platform will use an
+  /// initial size that satisfies the [preferredConstraints].
+  ///
+  /// If the [preferredSize] is not null and it does  not satisfy the
+  /// [preferredConstraints], then the platform will use an
+  /// initial size that does satisfy the [preferredConstraints] instead.
+  ///
+  /// If both [preferredSize] and [preferredConstraints] are null,
+  /// then the platform will use its own default size for the window.
+  ///
+  /// The [title] argument configures the window's initial title.
+  /// If omitted, some platforms might fall back to the app's name.
+  ///
+  /// The [delegate] argument can be used to listen to the window's
+  /// lifecycle. For example, it can be used to save state before
+  /// a window is closed.
+  ///
+  /// {@macro flutter.widgets.windowing.experimental}
+  @internal
+  factory RegularWindowController({
+    Size? preferredSize,
+    BoxConstraints? preferredConstraints,
+    String? title,
+    RegularWindowControllerDelegate? delegate,
+  }) {
+    if (!isWindowingEnabled) {
+      throw UnsupportedError(_kWindowingDisabledErrorMessage);
+    }
+
+    final WindowingOwner owner = WidgetsBinding.instance.windowingOwner;
+    final RegularWindowController controller = owner.createRegularWindowController(
+      delegate: delegate ?? RegularWindowControllerDelegate(),
+      preferredSize: preferredSize,
+      preferredConstraints: preferredConstraints,
+    );
+    if (title != null) {
+      controller.setTitle(title);
+    }
+    return controller;
+  }
+
+  /// Creates an empty [RegularWindowController].
+  ///
+  /// {@macro flutter.widgets.windowing.experimental}
+  // TODO(mattkae): Replace @internal with @visibleForTesting when this API is non-experimental
+  @internal
+  @protected
+  RegularWindowController.empty();
+
+  /// {@macro flutter.widgets.windowing.experimental}
+  @internal
+  @override
+  WindowType get type => WindowType.regular;
+
+  /// Request change to the content size of the window.
+  ///
+  /// The [size] describes the new requested window size. If the size disagrees
+  /// with the current constraints placed upon the window, the platform might
+  /// clamp the size within the constraiints.
+  ///
+  /// The platform is free to ignore this request.
+  ///
+  /// {@macro flutter.widgets.windowing.experimental}
+  @internal
+  void setSize(Size size);
```

Comment:
For the APIs where the platform might not honor the request, should we use the naming convention `requestFoo` instead of `setFoo`? For example, `requestSize`, `requestConstraints`, `requestTitle`, etc...